pkg/controllers/monitoring-stack/controller.go

```diff
@@ -100,7 +110,12 @@ func RegisterWithManager(mgr ctrl.Manager, opts Options) error {
 		Owns(&rbacv1.RoleBinding{}).WithEventFilter(p).
 		Owns(&monv1.ServiceMonitor{}).WithEventFilter(p).
 		Owns(&policyv1.PodDisruptionBudget{}).WithEventFilter(p).
-		Complete(r)
+		Watches(&source.Kind{Type: &grafanav1alpha1.GrafanaDataSource{}}, handler.EnqueueRequestsFromMapFunc(func(object client.Object) []reconcile.Request {
```

Comment:
nice!

Reply:
This does fail for me on a new cluster ... 
```
2021-12-14T12:31:27.489+1000    INFO    grafana-operator        Creating Grafana OperatorGroup
2021-12-14T12:31:27.489+1000    ERROR   controller-runtime.manager.controller.monitoringstack   Could not wait for Cache to sync        {"reconciler group": "monitoring.rhobs", "reconciler kind": "MonitoringStack", "error": "failed to wait for monitoringstack caches to sync: no matches for kind \"GrafanaDataSource\" in version \"integreatly.org/v1alpha1\""}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /home/sthaha/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:195
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /home/sthaha/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:221
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
        /home/sthaha/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/manager/internal.go:696
2021-12-14T12:31:27.489+1000    INFO    grafana-operator.controller.grafana-operator    Shutdown signal received, waiting for all workers to finish
2021-12-14T12:31:27.490+1000    ERROR   grafana-operator        End Reconcile - creation or updation error      {"req": "/", "error": "context canceled"}
```

Reply:
I dug around a bit ... 
https://github.com/rhobs/monitoring-stack-operator/pull/100/files may be a way to solve it. What I wanted to do  was to add a watch dynamically.  

see also: https://github.com/kubernetes-sigs/controller-runtime/issues/246